### PR TITLE
HTML Cheat Sheet

### PR DESCRIPTION
**Pull request by jsstrn**
**What does your Instant Answer do?**

It lists the HTML elements.

**What is the data source for your Instant Answer? (Provide a link if possible)**

[HTML element reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Element) from Mozilla Developer Network.

**Why did you choose this data source?**

I chose this because they listed HTML elements in a logical structure, alphabetized, and separated deprecated elements.

**Are there any other alternative (better) data sources?**

Others that I've looked at contained deprecated elements or only listed common HTML elements. There's a lot of outdated sources out there.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

Web developers. People learning HTML.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

Yes. See [here](https://duck.co/ideas/idea/6160/html-cheat-sheet).

**Are you having any problems? Do you need our help with anything?**

All is well. I'm definitely open to feedback and suggestions for improvement.

**Where did you hear about DuckDuckHack? (For first time contributors)**

TL;DR: Serendipity.

I've been a long time user of DuckDuckGo and found out about the DuckDuckHack open source projects after reading about its privacy policy and the "Internet bubble effect".

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

screen shot 2015-08-30 at 2 45 41 am

screen shot 2015-08-30 at 2 46 00 am

IA Page: https://duck.co/ia/view/html_cheat_sheet

### CODE REVIEW

**Comment by MariagraziaAlastra** [comment]
Hey @jsstrn, thanks for your contribution!
We'll review this and get back to you soon :)

File: share/goodie/cheat_sheets/json/html.json [hunk 1]
```diff
@@ -0,0 +1,341 @@
+{
+  "id": "html_cheat_sheet",
+  "name": "HTML Elements",
+  "description": "",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/html.json]: You can remove this, or fill it out to describe the cheat sheet

Comment (jsstrn) [inline share/goodie/cheat_sheets/json/html.json]: I've removed it.

File: share/goodie/cheat_sheets/json/html.json [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    "Table content",
+    "Forms",
+    "Embedded content",
+    "Scripting"
+  ],
+  "sections": {
+    "Document metadata": [
+      {
+        "key": "<head>",
+        "val": "The HTML Head Element (<head>) provides general information (metadata) about the document, including its title and links to or definitions of scripts and style sheets"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/html.json]: I think this cheat sheet is a little too verbose. Te idea is to get the point across pretty quickly and somewhat bluntly. We can reduce the bulk of text be removing the content that repeats in all of these. i.e. "The HTML X Element (<X>) provides"

As well, let's just skip to the important info and be more succint."Provides metadata about the document including title, scripts and stylesheets".

The purpose of the "More at" link is to give the user more details and depth, so they can visit the link if they really want to know things in more detail.

File: share/goodie/cheat_sheets/json/html.json [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+      {
+        "key": "<th>",
+        "val": "The HTML Table Header Cell Element (<th>) defines a cell that is a header for a group of cells of a table. The group of cells that the header refers to is defined by the scope and headers attribute."
+      },
+      {
+        "key": "<thead>",
+        "val": "The HTML Table Head Element (<thead>) defines a set of rows defining the head of the columns of the table."
+      },
+      {
+        "key": "<tr>",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/html.json]: For the case where the element name isn't self explanatory we can do something like "Table Row. Defines a row of cells in a table."

File: share/goodie/cheat_sheets/json/html.json (hunk 2 above)
Comment (jsstrn) [inline share/goodie/cheat_sheets/json/html.json]: I see your point. I'll get to work on it. Thank you.

File: share/goodie/cheat_sheets/json/html.json (hunk 3 above)
Comment (jsstrn) [inline share/goodie/cheat_sheets/json/html.json]: Good call.

File: share/goodie/cheat_sheets/json/html.json [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        "key": "<canvas>",
+        "val": "The HTML <canvas> Element can be used to draw graphics via scripting (usually JavaScript). For example, it can be used to draw graphs, make photo compositions or even perform animations. You may (and should) provide alternate content inside the <canvas> block. That content will be rendered both on older browsers that don't support canvas and in browsers with JavaScript disabled."
+      },
+      {
+        "key": "<noscript>",
+        "val": "The HTML <noscript> Element defines a section of html to be inserted if a script type on the page is unsupported or if scripting is currently turned off in the browser."
+      },
+      {
+        "key": "<script>",
+        "val": "     The HTML Script Element (<script>) is used to embed or reference an executable script within an HTML or XHTML document."
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/html.json]: Remove the spaces at the beginning?

**Comment by jagtalon** [comment]
Looks really useful!

**Comment by jsstrn** [comment]
Improved this cheat sheet by rewording the element descriptions to make them less verbose. Also removed some elements that are not as widely used.

Thanks for the suggestions, guys! It's been a great first experience for me! Woot! More to come! 

Updated screenshots:
screen shot 2015-08-30 at 4 38 48 pm

screen shot 2015-08-30 at 4 39 03 pm

**Comment by moollaza** [comment]
@jsstrn this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

BTW, for the future, we use 4-space indents :)

**Comment by moollaza** [comment]
@jsstrn congrats! It's live: https://duckduckgo.com/?q=html%20cheat%20sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by moollaza** [comment]
By the way, we think you did a great job with this! Are you interested in helping others who build similar Instant Answers in the future?

If yes, we'll reach out and add you to a private list of mentors as well as send over some DDG merch as an additional thank you. Just email <open@duckduckgo.com> with the subject 'I want to help.' Please include your shipping address, phone-number (required for shipping outside USA) and t-shirt size. We'll reach out from there!

Otherwise, thank you for this amazing contribution. We hope you'll keep exploring and playing around with DuckDuckHack.
